client/layout/logged-out.jsx

```diff
@@ -111,8 +112,13 @@ LayoutLoggedOut.propTypes = {
 
 export default connect( state => {
 	const section = getSection( state );
+	const currentRoute = getCurrentRoute( state );
+	const noMasterbarForRoute = currentRoute === '/log-in/jetpack';
+	const noMasterbarForSection = 'signup' === section.name || 'jetpack-connect' === section.name;
+
 	return {
-		masterbarIsHidden: ! masterbarIsVisible( state ) || 'signup' === section.name,
+		masterbarIsHidden:
+			! masterbarIsVisible( state ) || noMasterbarForSection || noMasterbarForRoute,
```

Comment:
Should this logic go inside the `masterbarIsVisible` selector?

Reply:
I wouldn't move it there, because `masterbarIsVisible` relies on the current UI state, and it is changeable. I'd like to keep this change contained for this use case only. Also, we'd need to test all of signup, which I really wouldn't want to do - this is supposed to affect only the Jetpack Connect flows.